### DISCUSSION TITLE
SRP Client Key and name conflicts

### DISCUSSION THREAD

**Discussion by agalliazzo** [openthread/openthread#8864] [Q&A] [answered]
Hi guys,
As far as I understood the SRP server keeps a sort of map with names and keys. If a client ask to renew its IP address the key should match with the one on the server else the update is refused.

Client key should be stored on non-volatile memory and be automatically loaded on reboot (excluding a device hard-reset of course). It seems that in my case (STM32WB stack) persistent data is not saving the key and when power goes down, at the next boot, I get
```
Mar 14 15:31:30 localhost otbr-agent[3677739]: 1d.19:20:48.902 [I] SrpServer-----: Received DNS update from fd4d:8b3c:e590:25db:5dc6:2a41:db55:733a
Mar 14 15:31:30 localhost otbr-agent[3677739]: 1d.19:20:48.903 [W] SrpServer-----: Name conflict: host name 70BE540205E18000.default.service.arpa. has already been allocated
Mar 14 15:31:30 localhost otbr-agent[3677739]: 1d.19:20:48.903 [W] SrpServer-----: Failed to process DNS Update section: Duplicated
Mar 14 15:31:30 localhost otbr-agent[3677739]: 1d.19:20:48.903 [W] SrpServer-----: Send fail response: 6
```

my code to send service to the server is the following, and of course, it works when it is firstly registered to the server
```C
		otSrpClientSetCallback(NULL, _CP_SrpCallback, ci);
		error = otSrpClientSetHostName(NULL, ci->EUI64);
		error = otSrpClientAddService(NULL, &service);
		error = otSrpClientSetHostAddresses(NULL, &otLocalMLEIDAddress, 1);
		error = otSrpClientStart(NULL, &SRP);
```

So my question is: is there any way to get the key from the openthrea API and later load it to the SRP Client? Or any way to manipulate the key?

**Comment by jwhui** [comment, marked as answer]
The Thread Specification requires certain information to be stored in non-volatile (including the SRP Client key) - see OpenThread's platform requirements in [include/openthread/platform/settings.h](https://github.com/openthread/openthread/blob/main/include/openthread/platform/settings.h). If the Thread device is not able to store this information in non-volatile storage, then there will be other issues as well.

**Reply by agalliazzo** [reply]
Thanks for the fast reply, I'll try to ask on the STM32 comunity about that hoping someone at ST will help me solve this problem.
